fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -153,6 +155,25 @@ public class TypeCoercionUtils {
 
     private static final Logger LOG = LogManager.getLogger(TypeCoercionUtils.class);
 
+    /**
+     * ensure the result's data type equals to the originExpr's dataType
+     */
+    public static Expression ensureSameResultType(
+            Expression originExpr, Expression result, ExpressionRewriteContext context) {
+        DataType originDataType = originExpr.getDataType();
+        DataType newDataType = result.getDataType();
+        if (originDataType.equals(newDataType)) {
+            return result;
+        }
+        // backend can direct use all string like type without cast
+        if (originDataType.isStringLikeType() && newDataType.isStringLikeType()) {
+            return result;
+        }
+        return FoldConstantRuleOnFE.PATTERN_MATCH_INSTANCE.visitCast(
```

Comment:
why use FoldConstantRuleOnFE here?

Reply:
for fast fold this cast

Reply:
functio name should related to constant folding

Reply:
> functio name should related to constant folding

`ensureSameResultTypeAndTryFoldCast` is too long